types/lodash/common/object.d.ts

```diff
@@ -1021,41 +1021,106 @@ declare module "../index" {
         functionsIn(): CollectionChain<string>;
     }
 
-    type GetIndexedField<T, K> = K extends keyof T
-        ? T[K]
-        : K extends `${number}`
-            ? 'length' extends keyof T
-                ? number extends T['length']
-                    ? number extends keyof T
-                        ? T[number]
-                        : undefined
-                    : undefined
-                : undefined
-            : undefined;
+    type GetFieldTypeOfArrayLikeByKey<T extends unknown[], K> =
+        K extends number ? T[K]
+        : K extends `${infer N extends number}` ? T[N]
+        : K extends keyof T ? T[K] : undefined;
 
-    type FieldWithPossiblyUndefined<T, Key> =
-        | GetFieldType<Exclude<T, undefined>, Key>
-        | Extract<T, undefined>;
+    type GetFieldTypeOfStringByKey<T extends string, K> =
+        K extends number ? T[K]
+        : K extends `${infer N extends number}` ? T[N]
+        : K extends keyof T ? T[K]
+        : undefined;
 
-    type IndexedFieldWithPossiblyUndefined<T, Key> =
-        | GetIndexedField<Exclude<T, undefined>, Key>
-        | Extract<T, undefined>;
+    type GetFieldTypeOfNarrowedByKey<T, K> =
+        T extends unknown[] ? GetFieldTypeOfArrayLikeByKey<T, K>
+        : T extends string ? GetFieldTypeOfStringByKey<T, K>
+        : K extends keyof T
+            ? K extends Symbol ? T[K]
+            : number extends keyof T ? T[K] | undefined
+            : string extends keyof T ? T[K] | undefined
```

Comment:
I wonder if `|undefined` should be removed from here, too, to avoid the breaking change seen on line 5390 of the tests? Same with the line above. Then this whole block could be simplified to:

`: K extends keyof T ? T[K]`

...which seems like the most reasonable thing to do.

Reply:
@aj-r this reads right, I did the update, I guess you need to re-approve the merge again.